### PR TITLE
Add a new pattern

### PR DESCRIPTION
**Pull request by lyriccoder**
New pattern has been added. It finds all method chaining
https://github.com/yegor256/aibolit/issues/49

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#50](https://github.com/yegor256/aibolit/issues/50) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT5UB7XAP/8111df3b-0efc-4265-8934-f32baec46b57); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: test/patterns/method_chaining/test_method_chaining.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+from unittest import TestCase
+from aibolit.patterns.method_chaining.method_chaining import MethodChainFinder
+from pathlib import Path
+
+
+class TestConcatString(TestCase):
```
Comment (paulodamaso) [inline test/patterns/method_chaining/test_method_chaining.py]: @lyriccoder Shouldn't this class be named `TestMethodChain`?

File: aibolit/patterns/method_chaining/method_chaining.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import javalang
+
+from aibolit.utils.utils import remove_comments
+
+
+class MethodChainFinder:
```
Comment (paulodamaso) [inline aibolit/patterns/method_chaining/method_chaining.py]: @lyriccoder OO classes souldn't have names ending in -er. Please take a look at [this article](https://www.yegor256.com/2015/03/09/objects-end-with-er.html)

File: aibolit/patterns/method_chaining/method_chaining.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        Takes path to java class file and returns AST Tree
+        :param filename:
+        :return: Tree
+        """
+        with open(filename, encoding='utf-8') as file:
+            res = javalang.parse.parse(remove_comments(file.read()))
+        return res
+
+    # flake8: noqa: C901
+    def value(self, filename: str):
```
Comment (paulodamaso) [inline aibolit/patterns/method_chaining/method_chaining.py]: @lyriccoder I coudn't figure out what this method is returning. Please add some documentation to it

File: aibolit/utils/utils.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import re
+
+
+def remove_comments(string):
```
Comment (paulodamaso) [inline aibolit/utils/utils.py]: @lyriccoder We shouldn't use utility classes. How bout creating a decorator which remove the comments of a string or file?

**Comment by 0crat** [comment]
Job was finished in 23 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)[/z](https://www.0crat.com/u/paulodamaso) didn't make CR comments

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
Pull request [#50](https://github.com/yegor256/aibolit/issues/50) was not merged, no payment for `ARC`, see [§28](http://www.zerocracy.com/policy.html#28)

**Pull request closed without merging**